src/nimony/semtypes.nim

```diff
@@ -623,6 +626,8 @@ proc semLocalTypeImpl(c: var SemContext; n: var Cursor; context: TypeDeclContext
         skipParRi n
       elif xkind == TupX:
         semTupleType c, n
+      elif xkind == TupConstrX:
+        semTupleType(c, n, isTupConstr = true)
```

Comment:
I think you should instead give the `is` operator more special logic rather than infesting `semTupleType` with this logic.